app/src/main/java/com/android/unio/ui/association/AssociationProfile.kt

```diff
@@ -281,13 +283,21 @@ private fun AssociationProfileContent(
     mutableStateOf(user!!.followedAssociations.contains(association!!.uid))
   }
   var enableButton by remember { mutableStateOf(true) }
+  val isConnected = Utils.checkInternetConnection(context)
+
   val onFollow = {
-    enableButton = false
-    associationViewModel.updateFollow(association!!, user!!, isFollowed) {
-      userViewModel.refreshUser()
-      enableButton = true
+    if (isConnected) {
```

Comment:
Nice fix to check whether the user is connect or not to ensure correct following tracking